lib/posix/posix.nim

```diff
@@ -1580,11 +1580,14 @@ else:
 
 when defined(macosx):
   # We can't use the NOSIGNAL flag in the ``send`` function, it has no effect
+  # Instead we should use SO_NOSIGPIPE in setsockopt
   var
     MSG_NOSIGNAL* = 0'i32
+    SO_NOSIGPIPE* {.importc, header: "<sys/socket.h>".}: cint
 else:
   var
     MSG_NOSIGNAL* {.importc, header: "<sys/socket.h>".}: cint
+    SO_NOSIGPIPE* = 0'i32
```

Comment:
Make it a const if = 0'i32 or even better remove it for when not defined(macosx). It's macosx specific.